### PR TITLE
Support reading rows from a Materialized View

### LINKED ISSUE #1324: Support Materialized Views in REST API
Instead of only grabbing tables from the keyspace also look at materialized views.

### PR DESCRIPTION
**Pull request by dougwettlaufer**
**What this PR does**:
Adds support for Materialized Views when reading rows in the REST API

**Which issue(s) this PR fixes**:
Closes #1324 

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated

### CODE REVIEW

File: restapi/src/main/java/io/stargate/web/restapi/dao/RestDB.java [hunk 1]
```diff
@@ -99,6 +100,41 @@ public Table getTable(String keyspaceName, String table) {
...
+   * Method for trying to find specific table or materialized view that exists in given keyspace.
+   * Keyspace must exist for call to work; otherwise {@link NotFoundException} will be thrown
+   *
+   * @param keyspaceName Name of keyspace to look for tables (must exist)
+   * @param table Name of table or materialized view to look for (must exist)
+   * @return Metadata for Table requested
+   * @throws NotFoundException If no keyspace with given name exists in the underlying data store,
+   *     or if no table with specified name exists within that keyspace.
+   */
+  public Table getTableForRead(String keyspaceName, String table) {
```
[the remark this answers is not in the document]
Comment (mpenick) [reply restapi/src/main/java/io/stargate/web/restapi/dao/RestDB.java:113]: Do we need to enforce that here? If you try to write to a MV then the underlying persistence with tell you can't

File: restapi/src/main/java/io/stargate/web/restapi/dao/RestDB.java [hunk 2]
```diff
@@ -77,24 +78,29 @@ public RestDB(
   }
 
   /**
-   * Method for trying to find specific table that exists in given keyspace. Keyspace must exist for
-   * call to work; otherise {@link NotFoundException} will be thrown
+   * Method for trying to find specific tableor materialized view that exists in given keyspace.
```
Comment (ivansenic) [inline restapi/src/main/java/io/stargate/web/restapi/dao/RestDB.java:81]: `tableor` missing space..

**Review by ivansenic — APPROVED** [review]
Clean and easy :+1:

EDIT: Completely forgot to check the side-effects, but thankful there's @mpenick with all eyes open :+1:

**Comment by tatu-at-datastax** [comment]
Do actual queries look similar (or identical) to queries against regular tables? If so, should "just work" with SGv2 I think, wrt Row access. But sounds like it might need some extra help for Column/Table metadata access?

**Review by ivansenic — APPROVED** [review]
This all seems fine to me...

**Pull request merged**